src/main/java/com/revature/controllers/TripController.java

```diff
@@ -1,137 +1,247 @@
+
+		return ResponseEntity.noContent().build();
+	}
+
+	/**
+	 * HTTP POST method (/trips)
+	 * @param trip represents the new Trip object being sent.
+	 * @return The newly created object along with a 201 code.
+	 */
+	@ApiOperation(value = "Adds a new trip")
+	@PostMapping
+	public ResponseEntity<TripDTO> addTrip(@Valid @RequestBody TripDTO tripDTO) {
+		Optional<Trip> newTrip = tripService.getTripById(tripDTO.getTripId());
+
+		if (newTrip.isPresent()) return ResponseEntity.badRequest().build();
+
+		return ResponseEntity.status(201).body(new TripDTO(tripService.addTrip(new Trip(tripDTO))));
+	}
+
+	/**
+	 * HTTP PUT method (/trips)
+	 *
+	 * @param trip represent the updated Trip object being sent.
+	 * @return The newly updated Trip object.
+	 */
+	@ApiOperation(value = "Updates a trip by ID", tags = {"Trip"})
+	@PutMapping
+	public ResponseEntity<TripDTO> updateTrip(@Valid @RequestBody TripDTO tripDTO) {
+		Optional<Trip> existingTrip = tripService.getTripById(tripDTO.getTripId());
+
+		if (existingTrip.isPresent()) return ResponseEntity.status(201).body(new TripDTO(tripService.updateTrip(new Trip(tripDTO))));
+
+		return ResponseEntity.badRequest().build();
+	}
+
+	/**
+	 * HTTP DELETE method (/trips/{id})
+	 *
+	 * @param id represents the Trip's ID.
+	 * @return A string that says which Trip was deleted.
+	 */
+	@ApiOperation(value = "Deletes a trip by ID", tags = {"Trip"})
+	@DeleteMapping("/{id}")
+	public ResponseEntity<String> deleteTripById(@PathVariable("id") int id) {
+		Optional<Trip> trip = tripService.getTripById(id);
+
+		if (trip.isPresent()) {
+			String message = tripService.deleteTripById(id);
+
+			return ResponseEntity.status(HttpStatus.ACCEPTED).body(message);
+		}
+
+		return ResponseEntity.notFound().build();
+	}
+
+	/**
+	 * HTTP GET method (/trips/rider/)
+	 * 
+	 * @param riderId
+	 * @param offset
+	 * @return A list of trips matching the rider's ID.
+	 */
+	@ApiOperation(value = "Return all trips", tags = {"Trip"})
+	@GetMapping("/rider")
+	public ResponseEntity<List<TripDTO>> getTripsByRider(
+			@RequestParam(name = "riderId", required = false) Integer riderId,
+			@RequestParam(name = "offset", required = false) Integer offset
+			){
+		List<TripDTO> trips = null;
+		if (offset != null) {
+			trips = tripService.getTripsByRiderIdDTO(riderId, offset);
+		}else {
+			trips = tripService.getTripsByRiderIdDTO(riderId);
+		}
+		if (trips != null && !trips.isEmpty()) return ResponseEntity.ok(trips);
+
+		return ResponseEntity.noContent().build();
+	}
+
+	/**
+	 * HTTP POST method (/trips/rider)
+	 * 
+	 * @param tripId represents the id of the trip the person is joining
+	 * @param rider represents the User joining the trip.
+	 * @return The newly updated Trip object.
+	 */
+	@ApiOperation(value = "Adds a user to a trip", tags = {"Rider", "Trip"})
+	@PostMapping("/rider")
+	public ResponseEntity<TripDTO> updateTripRider(
+			@RequestParam(name = "tripId") Integer tripId,
```

Comment:
These should be set to required as well, unless that is the default, in which case this can be resolved automatically.